### DISCUSSION TITLE
Possible bug: DataFetcherInputArgumentSchemaMismatchException incorrectly thrown?

### DISCUSSION THREAD

**Discussion by ddoongoor** [Netflix/dgs-framework#1671] [Q&A] [answered]
Hello,

After upgrading from Netflix DGS 7.20 to 7.30, I started to receive this exception:

```
 @InputArgument(name = "pagination") defined in class com.graphql.resolvers.BasicResolver in method `getItems` on parameter named `pagination` has no matching argument with name `pagination` in the GraphQL schema. No arguments on the field are defined in the schema.
```

but if I look in my schema, it really is there:

```
items(projectId: ID!, itemId: ID!, pagination: PaginationInput): Items!
```

Any ideas on what this could be?

Thanks

**Comment by srinivasankavitha** [comment]
I don't see any changes specifically between those 2 releases that could affect `@InputArgument` functionality. Could you also try with the latest release version `7.6.0`? 

Also, what does `PaginationInput` look like, so we can try to reproduce on our end.

**Reply by ddoongoor** [reply]
I tried it with `7.6.0` and I still see the error. The only difference I see between `7.2.0` and `7.3.0+` is the [introduction](https://github.com/Netflix/dgs-framework/pull/1565) of input argument checking.

`PaginationInput` is defined like this:

```
input PaginationInput {
  cursor: String
  limit: Int
}
```

**Comment by srinivasankavitha** [comment]
I see, ok. Thanks. Will try reproducing the issue on our end and see what's going on.

**Reply by srinivasankavitha** [reply]
What does your data fetcher method look like? I am not able to reproduce locally with that schema.

**Reply by ddoongoor** [reply]
Hello, my data fetcher looks something like this:

```Java
@DgsData(
    parentType = "parent",
    field = "items"
  )
  public Items getItems(
    final DataFetchingEnvironment dfe,
    @InputArgument final PaginationInput pagination
  ) {
    var item = dfe.getSource();
    var projectId = item.getProject().getId();
    var itemId = item.getId();

    return itemService.getItems(projectId, itemId, pagination);
  }
```

**Comment by ddoongoor** [comment]
Bump? Any updates?

**Reply by srinivasankavitha** [reply, marked as answer]
Ok, but `items` is a query type, correct? So your parent type need to be `Query`.
i.e.
```
@DgsData(
    parentType = "Query",
    field = "items"
  )
  public Items getItems(
    final DataFetchingEnvironment dfe,
    @InputArgument final PaginationInput pagination
  ) {
    var item = dfe.getSource();
    var projectId = item.getProject().getId();
    var itemId = item.getId();

    return itemService.getItems(projectId, itemId, pagination);
  }
```

**Reply by ddoongoor** [reply]
Not exactly what my issue was, but your response made me realize that the error is actually correct - I was adding a parameter to a field resolver which didn't belong.  Thanks!
